arbitrary-users-patch/build_images.sh

```diff
@@ -23,20 +23,29 @@ TAG=${TAG:-${DEFAULT_TAG}}
 NAME_FORMAT="${REGISTRY}/${ORGANIZATION}"
 
 PUSH_IMAGES=false
-if [ "$1" == "--push" ]; then
+if [ "$1" == "--push" ] || [ "$2" == "--push" ]; then
   PUSH_IMAGES=true
 fi
 
+RM_IMAGES=false
+if [ "$1" == "--rm" ] || [ "$2" == "--rm" ]; then
+  RM_IMAGES=true
+fi
+
 BUILT_IMAGES=""
 while read -r line; do
   base_image_name=$(echo "$line" | tr -s ' ' | cut -f 1 -d ' ')
   base_image=$(echo "$line" | tr -s ' ' | cut -f 2 -d ' ')
   echo "Building ${NAME_FORMAT}/${base_image_name}:${TAG} based on $base_image ..."
   docker build -t "${NAME_FORMAT}/${base_image_name}:${TAG}" --no-cache --build-arg FROM_IMAGE="$base_image" "${SCRIPT_DIR}"/ | cat
```

Comment:
what if we use docker buildx ? AFAIK there is no local registry publication in that case

Reply:
maybe when multiarch build will be implemented again, the one that works that is, we shall see if this option will be needed.

Reply:
buildx can be used for a single arch

Reply:
that's a problem for IBM PEs to solve :) 

perhaps we have to run each container build in a separate thread / action step / container so that they're not running out of disk. We barely have enough disk for the sidecars for x64 + 3-4 arches of the devfile registry, so when we're building the dozen sidecars x 3-4 arches we'll definitely be out of disk super fast.

Reply:
"the tool" being a GH action instead of the shell script that can be run locally? 

Yeah, that's great until you're stuck waiting for a GH action to fire for 5hrs (che-theia) and can't build locally. :) 

But conceptually, yes, I agree. reusing GH actions instead of writing our own wrappers is a noble cause. As long as it solves our dogfooding/development workflows.

Reply:
+1 for using buildx, we will eventually switch to that for multi-arch anyway

Reply:
> "the tool" being a GH action instead of the shell script that can be run locally?

 no, I mean the tool = `docker buildx` not `docker build` (it can run locally as well)
just that build, publish, delete, better caching options is already managed, no need to add new .sh scripts around that

Reply:
> IMHO it looks cleaner to rely on the tools than do .sh wrappers to handle publish, remove, etc when the tool can handle both publish and remove by itself

Right, but then when we want to call the same set of tool commands for both nightly and release builds, we can either:
* use the shell script wrappers, or
* copy the same docker commands boilerplate into two github action scripts.

Why not collect the commands into a single shell script for portability? Then you can run it:
* as a nightly
* as a release
* locally